src/client/src/services/limitChecker/limitChecker.openfin.ts

```diff
@@ -1,128 +1,56 @@
-/* eslint-disable @typescript-eslint/ban-ts-comment */
-// This file uses legacy OpenFin API which is not compatible with the
-// types of the version of the OpenFin lib we are using. This should
-// be fixed, but in the meantime, we are ignoring type checking on
-// on this file.
-// @ts-nocheck
 import { bind } from "@react-rxjs/core"
-import { BehaviorSubject, of, race, Subject, timer } from "rxjs"
-import { map, switchMap, take, tap } from "rxjs/operators"
+import { BehaviorSubject, race, Subject, timer } from "rxjs"
+import { map, take, tap } from "rxjs/operators"
 
 import { checkLimitFn } from "./types"
 
-const limitCheckSubscriptionUuid$ = new BehaviorSubject<string | null>(null)
+window.fdc3.addContextListener("limit-checker-status", (context) => {
+  if (context.id) limitCheckSubscription$.next(context.id.isAlive)
+})
+
+const limitCheckSubscription$ = new BehaviorSubject<string>("false")
 
 export const [useIsLimitCheckerRunning] = bind<boolean>(
-  limitCheckSubscriptionUuid$.pipe(map((uuid) => uuid !== null)),
+  limitCheckSubscription$.pipe(map((isAlive) => isAlive === "true")),
 )
 
-const setLimitCheckSubscriber = (uuid: string, topic: string) => {
-  if (topic === REQUEST_LIMIT_CHECK_TOPIC) {
-    console.info(LOG_NAME, `${uuid} has subscribed as a limit checker`)
-    limitCheckSubscriptionUuid$.next(uuid)
-  }
-}
-
-const removeLimitCheckSubscriber = (uuid: string, topic: string) => {
-  if (topic === REQUEST_LIMIT_CHECK_TOPIC) {
-    console.info(LOG_NAME, `${uuid} has unsubscribed as a limit checker`)
-    limitCheckSubscriptionUuid$.next(null)
-  }
-}
-
-const REQUEST_LIMIT_CHECK_TOPIC = "request-limit-check"
-const LIMIT_CHECKER_UUID = "LIMIT-CHECKER"
-const LIMIT_CHECKER_STATUS_TOPIC = "request-limit-check-status"
 const LOG_NAME = "OpenFin: "
-let limitCheckId = 1
+
+let limitCheckId = 0
 
 export const checkLimit$: checkLimitFn = (message: {
   tradedCurrencyPair: string
   notional: number
   rate: number
-}) =>
-  limitCheckSubscriptionUuid$.pipe(
-    take(1),
-    switchMap((uuid) => {
-      if (uuid === null) {
-        console.info(
-          LOG_NAME,
-          "client side limit check not up, will delegate to server",
-        )
-        return of(true)
-      }
-
-      console.info(LOG_NAME, `checking if limit is ok with ${uuid}`)
-
-      const obs = new Subject<boolean>()
-
-      const limitCheckResponse = (msg: { result: boolean }) => {
-        console.info(LOG_NAME, `${uuid} limit check response was`, msg)
-        obs.next(msg.result)
-      }
-
-      const topic = `limit-check-response (${++limitCheckId})`
-
-      fin.desktop.InterApplicationBus.subscribe(uuid, topic, limitCheckResponse)
-
-      const payload = {
-        ...message,
-        id: limitCheckId,
-        responseTopic: topic,
-      }
-
-      fin.desktop.InterApplicationBus.send(
-        uuid,
-        REQUEST_LIMIT_CHECK_TOPIC,
-        payload,
-      )
+}) => {
```

Comment:
another nit - not you - all types should be capitalised .. `checkLimitFn` -> `CheckLimitStreamGenerator` or something